core/shared/src/main/scala/fs2/Pull.scala

```diff
@@ -68,6 +69,9 @@ final class Pull[+F[_], +O, +R] private (private val free: FreeC[Algebra[Nothing
   /** Applies the resource of this pull to `f` and returns the result in a new `Pull`. */
   def map[R2](f: R => R2): Pull[F, O, R2] = Pull.fromFreeC(get.map(f))
 
+  /** Applies the outputs of this pull to `f` and returns the result in a new `Pull`. */
+  def mapOutput[O2](f: O => O2): Pull[F, O2, R] = Pull.mapOutput(this)(f)
```

Comment:
If we can't figure out why `ResourceSafetySpec` fails when `Stream#map` is implemented in terms of `mapOutput`, then we'll at least need to make this function private or move it to the test class that uses it.